src/state.c

```diff
@@ -1237,7 +1246,9 @@ xkb_state_mod_names_are_active(struct xkb_state *state,
             ret = -1;
             break;
         }
-        wanted |= (1u << idx);
+        wanted |= is_real_mod(state->keymap, idx)
+            ? (1u << idx)
+            : state->keymap->mods.mods[idx].mapping;
```

Comment:
this screams for a little helper function where the name explains what it does

Reply:
Done